internal/generator/vector/output/splunk/splunk.go

```
@@ -98,13 +117,17 @@ func New(id string, o *adapters.Output, inputs []string, secrets observability.S
 		builder.WriteString(sourceTmpl)
 	}
 
-	builder.WriteString("\n._internal.splunk.sourcetype = \"_json\"\n")
-
 	if o.Splunk.PayloadKey != "" {
 		path := vectorhelpers.SplitPath(string(o.Splunk.PayloadKey))
 		quotedSegments := vectorhelpers.QuotePathSegments(path)
 		quotedPathArray := fmt.Sprintf("[%s]", strings.Join(quotedSegments, ","))
-		builder.WriteString(fmt.Sprintf(payloadKeyTmpl, quotedPathArray))
+		if o.Splunk.SourceType != "" {
+			builder.WriteString(fmt.Sprintf(payloadKeysourceTypeTmpl, quotedPathArray, commontemplate.TransformUserTemplateToVRL(o.Splunk.SourceType)))
```

golangci-lint (2.12.2)

[error] QF1012: Use fmt.Fprintf(...) instead of WriteString(fmt.Sprintf(...))

(staticcheck)